server/src/test/java/org/elasticsearch/snapshots/DedicatedClusterSnapshotRestoreIT.java

```diff
@@ -981,6 +990,123 @@ public void testRestoreShrinkIndex() throws Exception {
         ensureYellow();
     }
 
+    public void testSnapshotTotalAndIncrementalSizes() throws IOException {
+        Client client = client();
+        final String indexName = "test-blocks-1";
+        final String repositoryName = "repo-" + indexName;
+        final String snapshot0 = "snapshot-0";
+        final String snapshot1 = "snapshot-1";
+
+        createIndex(indexName);
+
+        int docs = between(10, 100);
+        for (int i = 0; i < docs; i++) {
+            client.prepareIndex(indexName, "type").setSource("test", "init").execute().actionGet();
+        }
+
+        logger.info("--> register a repository");
+
+        final Path repoPath = randomRepoPath();
+        assertAcked(client.admin().cluster().preparePutRepository(repositoryName)
+            .setType("fs")
+            .setSettings(Settings.builder().put("location", repoPath)));
+
+        logger.info("--> verify the repository");
+        VerifyRepositoryResponse verifyResponse = client.admin().cluster().prepareVerifyRepository(repositoryName).get();
+        assertThat(verifyResponse.getNodes().length, equalTo(cluster().numDataAndMasterNodes()));
+
+        logger.info("--> create a snapshot");
+        CreateSnapshotResponse snapshotResponse = client.admin().cluster().prepareCreateSnapshot(repositoryName, snapshot0)
+            .setIncludeGlobalState(true)
+            .setWaitForCompletion(true)
+            .execute().actionGet();
+        assertThat(snapshotResponse.status(), equalTo(RestStatus.OK));
+        ensureSearchable();
+
+        SnapshotsStatusResponse response = client.admin().cluster().prepareSnapshotStatus(repositoryName)
+            .setSnapshots(snapshot0)
+            .execute()
+            .actionGet();
+
+        List<SnapshotStatus> snapshots = response.getSnapshots();
+
+        List<Path> files = scanSnapshotFolder(repoPath);
+        assertThat(snapshots, hasSize(1));
+        SnapshotStats stats = snapshots.get(0).getStats();
+
+
+        assertThat(stats.getTotalFileCount(), is(files.size()));
+        assertThat(stats.getTotalSize(), is(files.stream().mapToLong(f -> {
+            try {
+                return Files.size(f);
+            } catch (IOException e) {
+                throw new RuntimeException(e.getMessage(), e);
```

Comment:
maybe just `throw new UncheckedIOException(e)`. No need to pass the message as an extra parameter, it's just tests.